lib/Items/ApplicationDockItem.vala

```diff
@@ -442,11 +442,12 @@ namespace Plank
 				
 				foreach (var s in actions) {
 					var values = actions_map.get (s).split (";;");
-					
 					var item = create_menu_item (s, values[1], true);
 					item.activate.connect (() => {
 						try {
-							AppInfo.create_from_commandline (values[0], null, AppInfoCreateFlags.NONE).launch (null, null);
+							var desktop_id = File.new_for_uri (Prefs.Launcher).get_basename ();
+							var app_info = new DesktopAppInfo (desktop_id);
+							app_info.launch_action (values[2], new AppLaunchContext ());
```

Comment:
THis should use an `Gdk.AppLaunchContext` with the timestamp from when the user clicked in the button. this allow the launched window steal the focus as intended.

Reply:
Tnx for this suggestion, I was not aware of the `Gdk.AppLaunchContext`, but the timestamp feature makes perfect sense!

I've switched to using it with [this commit](https://github.com/elementary/dock/pull/166/commits/4d30d2baa666c350c2c1d9febf6e89377a7899cb). I hope I've set the timestamp correctly. I did not find examples of this in other elementary projects. Do you think it would make sense to open a pull request for this in [the applications-menu project](https://github.com/elementary/applications-menu/search?q=AppLaunchContext) as well?

Reply:
> I've switched to using it with [this commit](https://github.com/elementary/dock/pull/166/commits/4d30d2baa666c350c2c1d9febf6e89377a7899cb). I hope I've set the timestamp correctly.

The code seems fine.

> I did not find examples of this in other elementary projects. Do you think it would make sense to open a pull request for this in [the applications-menu project](https://github.com/elementary/applications-menu/search?q=AppLaunchContext) as well?

Yes, that would be great.

Reply:
Cool, I opened a PR for it [here](https://github.com/elementary/applications-menu/pull/549).